src/Rings/ReesAlgebra.jl

```diff
@@ -0,0 +1,166 @@
+    f_dual = dual(f)
+    is_surjective(f_dual) || error("it can not be verified that the map is versal")
+  end
+
+  M = domain(f)
+  R = base_ring(M)
+  F = codomain(f)
+  R === base_ring(F) || error("modules must be defined over the same ring")
+  P = presentation(M)::ChainComplex
+  p = map(P, 0)
+  FM = P[0]
+  r = rank(FM)
+  sym_FM, s = PolynomialRing(R, [Symbol(var_name*"$i") for i in 0:r-1])
+  sym_F, t = PolynomialRing(R, [Symbol("t$i") for i in 1:rank(F)])
+  imgs = Vector{elem_type(sym_F)}()
+  for v in gens(FM)
+    w = coordinates(f(p(v)))
+    push!(imgs, sum(w[i]*t[i] for i in 1:length(t)))
+  end
+  sym_g = hom(sym_FM, sym_F, imgs)
+  K = kernel(sym_g)
+  rees, _ = quo(sym_FM, K)
+  return rees
+end
+
+function rees_algebra(M::FreeMod; var_name::String="s")
+  R = base_ring(M)
+  r = rank(M)
+  S_tmp, s = PolynomialRing(R, [Symbol(var_name*"$i") for i in 0:r-1])
+  S, _ = grade(S_tmp)
+  return S
+end
+
+function rees_algebra(M::SubQuo; var_name::String="s", check::Bool=true)
+  success, p, sigma = is_projective(M)
+  if success
+    # The easy case: The Rees algebra is simply a polynomial ring 
+    # modulo linear equations in the variables parametrized by the base.
+    R = base_ring(M)
+    r = ngens(M)
+    S_tmp, s = PolynomialRing(R, [Symbol(var_name*"$i") for i in 0:r-1])
+    S, sg = grade(S_tmp)
+    A = matrix(p) # The presentation matrix
+    I = ideal(S, [sum(A[i, j]*sg[j] for j in 1:length(s)) for i in 1:nrows(A)])
+    Q, sq = quo(S, I)
+    return Q
+  end
+
+  # The complicated case. We construct a versal morphism f : M → F to a 
+  # free module along the lines of Proposition 1.3 of [1] (in its published version).
+  return rees_algebra(_versal_morphism_to_free_module(M), check=false)
+end
+
+### Auxiliary methods needed for the Rees algebras
+
+### The following function is implemented along the lines of Proposition 1.3 of [1] in 
+# its published version.
+function _versal_morphism_to_free_module(M::SubQuo)
+  R = base_ring(M)
+  R1 = FreeMod(R, 1)
+  M_double_dual, psi = double_dual(M, cod=R1)
+  M_dual = domain(element_to_homomorphism(zero(M_double_dual)))
+  pres_M_dual = presentation(M_dual)::ChainComplex
+  g = map(pres_M_dual, 0) # The projection of a free module onto M_dual
+  g_dual = dual(g, cod=R1, codomain_dual=M_double_dual)
+  return compose(psi, g_dual)
+end
+
+function proj(S::MPolyRing_dec)
+  is_standard_graded(S) || error("ring must be standard graded")
+  return ProjectiveScheme(S)
+end
+
+function proj(Q::MPolyQuo{<:MPolyElem_dec})
+  S = base_ring(Q)
+  is_standard_graded(S) || error("ring must be standard graded")
+  return ProjectiveScheme(Q)
+end
+
+function is_injective(f::ModuleFPHom) 
+  return iszero(kernel(f)[1])
+end
+
+function is_surjective(f::ModuleFPHom)
+  return iszero(cokernel(f))
+end
+
+function is_isomorphism(f::ModuleFPHom)
+  return is_injective(f) && is_surjective(f)
+end
+
+### Auxiliary deflections for MPolyQuos to make arithmetic work in the Rees algebras
```

Comment:
@wdecker : 

Since you are working on the `MPolyQuo`s these days: For my purposes, it would be nice, if the code on `MPolyQuo`s was so generic that it would also work when the coefficient ring is not a field (resp. Singular-understandable). Do you think, it is possible, to also point out two or three basic methods that must be 
implemented to derive the remaining functionality in a generic way? 

For instance, these few methods below could then be sufficient. Imho that would be great!